src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -118,16 +120,64 @@ sealed class NumericRange[T](
     }
   }
 
+  private def indexOfTyped(elem: T, from: Int): Int = {
+    val pos = posOf(elem)
+    pos match {
+      case -2                 => super.indexOf(elem, from)
+      case pos if pos >= from => pos
+      case _                  => -1
+    }
+  }
```

Comment:
for style points, omit the val and prefer
```
posOf(elem) match {
}
```
which also eliminates the extra braces. Some people like `def f = posOf(elem) match {` style.